pkg/manifest/mantaray/marshal_test.go

```diff
@@ -201,3 +202,54 @@ func TestMarshal(t *testing.T) {
 	// 	}
 	// }
 }
+
+func Test_UnmarshalBinary(t *testing.T) {
+	decode := func(s string) []byte {
+		data, err := hex.DecodeString(s)
+		if err != nil {
+			t.Fatalf("expected no error, got %v", err)
+		}
+		return data
+	}
+
+	tests := []struct {
+		data []byte // node binary data
+		err  error  // tells if UnmarshalBinary should return error
+	}{
+		{
+			data: nil,
+			err:  ErrTooShort,
+		},
+		{
+			data: make([]byte, nodeHeaderSize-1),
+			err:  ErrTooShort,
+		},
+		{
+			data: make([]byte, nodeHeaderSize),
+			err:  ErrInvalidVersionHash,
+		},
+		{ // fork with vaild metadata (size is 93)
+			data: decode("00000000000000000000000000000000000000000000000000000000000000005768b3b6a7db56d21d1abff40d41cebfc83448fed8d7e9b06ec0d3b073f28f200000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000016012f0000000000000000000000000000000000000000000000000000000000e87f95c3d081c4fede769b6c69e27b435e525cbd25c6715c607e7c531e329639005d7b22776562736974652d696e6465782d646f63756d656e74223a2233356561656538316262363338303436393965633637316265323736326465626665346662643330636461646139303232393239646131613965366134366436227d0a"),
+			err:  nil,
+		},
+		{ // fork with metadata, where metadata size value exceeds actual metadata size (size is 95, want 93)
+			data: decode("00000000000000000000000000000000000000000000000000000000000000005768b3b6a7db56d21d1abff40d41cebfc83448fed8d7e9b06ec0d3b073f28f200000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000016012f0000000000000000000000000000000000000000000000000000000000e87f95c3d081c4fede769b6c69e27b435e525cbd25c6715c607e7c531e329639005f7b22776562736974652d696e6465782d646f63756d656e74223a2233356561656538316262363338303436393965633637316265323736326465626665346662643330636461646139303232393239646131613965366134366436227d0a"),
+			err:  ErrInvalidOutput,
+		},
+		{ // fork with metadata, where metadata size value exceeds actual metadata size (size is 96, want 93)
+			data: decode("00000000000000000000000000000000000000000000000000000000000000005768b3b6a7db56d21d1abff40d41cebfc83448fed8d7e9b06ec0d3b073f28f200000000000000000000000000000000000000000000000000000000000000000000000000080000000000000000000000000000000000000000000000000000016012f0000000000000000000000000000000000000000000000000000000000e87f95c3d081c4fede769b6c69e27b435e525cbd25c6715c607e7c531e32963900607b22776562736974652d696e6465782d646f63756d656e74223a2233356561656538316262363338303436393965633637316265323736326465626665346662643330636461646139303232393239646131613965366134366436227d0a"),
+			err:  ErrInvalidOutput,
+		},
+	}
+
+	for _, tc := range tests {
+		n := New()
+		err := n.UnmarshalBinary(tc.data)
+
```

Comment:
Since the test cases aren't named, executing every test case as a sub-test (`t.Run("", func(t *testing.T) {...`) will help to accurately identify the problematic test case.